src/generators/legacy-json/utils/buildSection.mjs

```diff
@@ -0,0 +1,543 @@
+import { unified } from 'unified';
+import html from 'remark-html';
+import {
+  DEFAULT_EXPRESSION,
+  LEADING_HYPHEN,
+  NAME_EXPRESSION,
+  PARAM_EXPRESSION,
+  RETURN_EXPRESSION,
+  TYPE_EXPRESSION,
+} from '../constants.mjs';
+import { buildHierarchy } from './buildHierarchy.mjs';
+
+const sectionTypePlurals = {
+  module: 'modules',
+  misc: 'miscs',
+  class: 'classes',
+  method: 'methods',
+  property: 'properties',
+  global: 'globals',
+  example: 'examples',
+  // Constructors should go under a class sections' signatures property
+  ctor: 'signatures',
+  classMethod: 'classMethods',
+  event: 'events',
+  var: 'vars',
+};
+
+/**
+ * @param {import('../types.d.ts').HierarchizedEntry} entry
+ * @returns {import('../types.d.ts').Meta | undefined}
+ */
+function createMeta(entry) {
+  const makeArrayIfNotAlready = val => (Array.isArray(val) ? val : [val]);
+
+  const { added_in, n_api_version, deprecated_in, removed_in, changes } = entry;
+  if (added_in || n_api_version || deprecated_in || removed_in) {
+    return {
+      changes,
+      added: added_in ? makeArrayIfNotAlready(added_in) : undefined,
+      napiVersion: n_api_version
+        ? makeArrayIfNotAlready(n_api_version)
+        : undefined,
+      deprecated: deprecated_in
+        ? makeArrayIfNotAlready(deprecated_in)
+        : undefined,
+      removed: removed_in ? makeArrayIfNotAlready(removed_in) : undefined,
+    };
+  }
+
+  return undefined;
+}
+
+/**
+ * @param {import('../types.d.ts').HierarchizedEntry} entry Section's AST entry
+ * @param {HeadingMetadataParent} head Head node of the entry
+ * @returns {import('../types.d.ts').Section}
+ */
+function createSection(entry, head) {
+  const text = textJoin(head.children);
+
+  return {
+    textRaw: text,
+    type: head.data.type,
+    name: text.toLowerCase().replaceAll(' ', '_'),
+    displayName: head.data.name,
+    meta: createMeta(entry),
+    introduced_in: entry.introduced_in,
+  };
+}
+
+/**
+ * @param {string} textRaw Something like `new buffer.Blob([sources[, options]])`
+ * @param {Array<import('../types.d.ts').List} values
+ * @returns {import('../types.d.ts').MethodSignature}
+ */
+function parseSignature(textRaw, values) {
```

Comment:
Can the core domain logic from these functions be extracted to utilities? To prevent this file from exploding in size?